extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java

```diff
@@ -227,16 +242,31 @@ private static void s2iBuild(KubernetesClient client, List<HasMetadata> buildRes
      * @param buildConfig The build config.
      * @param binaryFile The binary file.
      */
-    private static void s2iBuild(OpenShiftClient client, BuildConfig buildConfig, File binaryFile) {
-        Build build = client.buildConfigs().withName(buildConfig.getMetadata().getName()).instantiateBinary()
-                .fromFile(binaryFile);
+    private static void s2iBuild(OpenShiftClient client, BuildConfig buildConfig, File binaryFile, S2iConfig s2iConfig) {
+        Build build;
+        try {
+            build = client.buildConfigs().withName(buildConfig.getMetadata().getName()).instantiateBinary()
+                    .withTimeoutInMillis(s2iConfig.buildTimeout.toMillis()).fromFile(binaryFile);
+        } catch (Exception e) {
+            if (e.getCause() instanceof StreamResetException) {
+                LOG.warn("Stream was reset while building. Falling back to building with the 'oc' binary.");
+                if (!ExecUtil.exec("oc", "start-build", buildConfig.getMetadata().getName(), "--from-archive",
```

Comment:
Wondering if at some point we should get the `oc` client from the OpenShift instance. We had incompatibilities in the past and it would avoid having the user install `oc`.

Reply:
This is only a fallback that @iocanel added because he saw the client failing is some weird circumstances. I'll leave it up to him to decide

Reply:
Agree, I don't like the idea of having the user depend on any kind of binary. This is just a fallback mechanism to work around a known issue in the client.

Reply:
Never tried to grab the binary from the instance, I will have a look.